packages/rendermime/src/latex.ts

```diff
@@ -9,20 +9,86 @@
 // Other minor modifications are also due to StackExchange and are used with
 // permission.
 
+import {
+  IRenderMime
+} from '@jupyterlab/rendermime-interfaces';
+
 const inline = '$'; // the inline math delimiter
 
 // MATHSPLIT contains the pattern for math delimiters and special symbols
 // needed for searching for math in the text input.
 const MATHSPLIT = /(\$\$?|\\(?:begin|end)\{[a-z]*\*?\}|\\[{}$]|[{}]|(?:\n\s*)+|@@\d+@@|\\\\(?:\(|\)|\[|\]))/i;
 
-// A module-level initialization flag.
-let initialized = false;
-
 
 // Stub for window MathJax.
 declare var MathJax: any;
 
 /**
+ * The MathJax Typesetter.
+ */
+export
+class MathJaxTypesetter implements IRenderMime.ILatexTypesetter {
+  /**
+   * Construct the typesetter.
+   */
+  constructor() {
+    this._init();
```

Comment:
Perhaps not init here so we don't call this unless we actually render something.